File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ans =0;
        int stock=prices[0];
        for(auto& i : prices)
        {
            ans=max(ans,i-stock);
            stock=min(i,stock);
        }
        return ans;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int n=prices.length;
        int x=Integer.MAX_VALUE,ans=Integer.MIN_VALUE;
        for(int i=0;i<n;i++)
        {
            x=Math.min(x,prices[i]);
            ans=Math.max(ans,prices[i]-x);
        }
        return ans;
    }
}